workflow/metrics/collector.go

```diff
@@ -131,4 +156,77 @@ func (wc *workflowCollector) collectWorkflow(ch chan<- prometheus.Metric, wf wfv
 		addGauge(descWorkflowFinishedAt, float64(wf.Status.FinishedAt.Unix()))
 	}
 
+	// Collect Node metrics
+	nodes := wf.Status.Nodes
+	if wf.Status.IsOffloadNodeStatus() && wc.offloadNodeStatusRepoSupplier != nil {
+		offloadNodeStatusRepo := wc.offloadNodeStatusRepoSupplier.GetOffloadNodeStatusRepo()
+		if offloadNodeStatusRepo.IsEnabled() {
+			offloadedNodes, err := offloadNodeStatusRepo.Get(string(wf.UID), wf.GetOffloadNodeStatusVersion())
+			if err != nil {
+				log.Errorf("not able to get offloaded nodes")
```

Comment:
I don't think this is the right way to deal with this error - can we return on this line or something - I think following code would expect nodes to be set - but they might not be